feat(clayui.com): Improve headings & content structure of components

kresimir-coko opened the pull request:
Fixes #3211 

Hey @bryceosterhaus, I also added another commit in here. Components in question:
- Checkbox
- Management Toolbar
- Select Box
- Sticker
- Table
- Tabs

### API Table
It adds `false` in the API Table under `required` for optional props. This is a proposition and feel free to not merge it. I think it will make the table look less empty at the cost of visual clarity where the required ones are clearly distinct from the optional ones. 
Another idea is to show both true/false for required/optional props but add some color to differentiate between them.

### Tabs
I've reduced the inside content of the tabs here in order to streamline the example so it focuses on the usage instead of having a lot of content which muddied up the example

bryceosterhaus commented on packages/clay-form/CHECKBOX.mdx:
```diff
@@ -32,22 +41,36 @@ Use [`checked`](#api-checked) property to check or not the checkbox.
 
 <CheckboxLabel />
 
+### Inline Checkboxes
```
For all of these titles, I think adding in "Checkbox" is redundant since they are on the checkbox page. They could simply be `Container Props... Label... Inline...`

bryceosterhaus commented on packages/clay-form/CHECKBOX.mdx:
```diff
@@ -32,22 +41,36 @@ Use [`checked`](#api-checked) property to check or not the checkbox.
 
 <CheckboxIndeterminate />
 
+### Checkbox with a callback
```
This should likely be our first example since we expect users to always use these components in a controlled way

kresimir-coko commented:
Great comments @bryceosterhaus I was looking into what was already there and just improved it a bit. I like removing Clay prefixes and component names from the headings and navigation, makes it look more clear.

kresimir-coko commented:
I've made the requested changes @bryceosterhaus plus a bit more, still contained within the confines of the issue 😄 

One thing to note, the links to rows in the API table don't really work. For example `the [`position`](#api-position) property.` doesn't scroll down to the API table. We can open an issue to fix all of those, but before that what would be the correct sytax to get the links to work?

bryceosterhaus commented on clayui.com/content/docs/components/css-checkbox-radio.md:
```diff
@@ -24,15 +20,15 @@ lexiconDefinition: 'https://liferay.design/lexicon/core-components/forms/radio-c
 
 By default, any number of checkboxes and radios that are immediate sibling will be vertically stacked and appropriately spaced with `.form-check`.
 
+## Default Checkboxes
```
Since this is nested under "Default", we can avoid redundancy by just calling it "Checkboxes".

bryceosterhaus commented on clayui.com/content/docs/components/css-checkbox-radio.md:
````diff
@@ -74,6 +70,8 @@ By default, any number of checkboxes and radios that are immediate sibling will
 </div>
 ```
 
+## Default Radios
````
Redundancy

bryceosterhaus commented on clayui.com/content/docs/components/css-checkbox-radio.md:
```diff
@@ -24,15 +20,15 @@ lexiconDefinition: 'https://liferay.design/lexicon/core-components/forms/radio-c
 By [Bootstrap](https://getbootstrap.com/docs/4.1/components/forms/#checkboxes-and-radios)
 
-### Default (stacked)
+# Default
```
We shouldn't have multiple `<h1>` tags on a page. I think the main categories should be an `h2` via `## some heading` and then nested items would be an h3

bryceosterhaus commented on packages/clay-sticker/README.mdx:
```diff
@@ -14,30 +14,34 @@ import {
 
--   [User Icon](#user-icon)
+-   [Composing](#composing)
+    -   [Using Color](#using-color)
```
I don't think these would be nested under "Composing". I would also rename it from "Using Color" to "Display Type"

bryceosterhaus replied:
I think I would also drop the "Composing" name and use "Positioning". I would also place this part last since its likely the least consulted. Display type would be first

bryceosterhaus commented on packages/clay-tabs/README.mdx:
```diff
@@ -19,7 +21,7 @@ import {Tabs, TabsDropdown} from '$packages/clay-tabs/docs/index';
 </div>
 
-## Introduction
+# Composing
```
I wouldn't call this "Composing", its more of an "Introduction"

kresimir-coko commented:
Hey @bryceosterhaus my bad with the headings, didn't realize why we had them starting at h2 until you pointed it out.

I've removed the h1's and changed the wording like you suggested in this latest commit.

I'll open the issue regarding the links to the API table 👍
